public/js/src/module/controller-webform.js

```diff
@@ -591,23 +591,26 @@ function _saveRecord(survey, draft, recordName, confirmed) {
             _resetForm(survey, { isOffline: true });
 
             if (draft) {
+                return true;
+            }
+
+            return records.uploadQueue({ isUserTriggered: !draft });
+        })
+        .then((success) => {
+            if (success && draft) {
                 gui.alert(
                     t('alert.recordsavesuccess.draftmsg'),
                     t('alert.savedraftinfo.heading'),
                     'info',
                     5
                 );
-            } else {
+            } else if (!success && !draft) {
```

Comment:
Hi @eyelidlessness , we did notice one small regression! It looks like a 401 (non-auth) response would fall into this if statement and have its login prompt replaced by the "5 minute timer" GUI.

So for the scenario that a user is not logged in and attempts make a submission, their login prompt would flash for a second then it would be replaced by this message

Reply:
Here's an idea for working around this: https://github.com/jnm/enketo-express/commit/f881d5d9911a8a1edf84ac0fe754cf27f1679c1e. In general, when Enketo is using OpenRosa authentication, I don't think the login confirmation modal should disappear unless the user explicitly dismisses it. I wonder if there are any scenarios that I'm overlooking, though.

### Before linked commit

https://user-images.githubusercontent.com/2085013/231403057-cb8d02da-81bb-46b4-b43f-0959b73e358d.mp4

### After

https://user-images.githubusercontent.com/2085013/231403044-5971da8f-49b3-44b7-b63f-1d7cb0f73b5c.mp4